### PR TITLE
Change ServiceDescriptor annotation to Descriptor

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
$subject
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/3005
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/3038
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/3039
## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [x] Updated the spec

### CODE REVIEW

File: ballerina-tests/Ballerina.toml [hunk 1]
```diff
@@ -1,24 +1,24 @@
 [package]
 org = "ballerina"
 name = "grpc_tests"
-version = "@toml.version@"
+version = "1.3.1"
```
Comment (daneshk) [inline ballerina-tests/Ballerina.toml:4]: is this intentional?

File: ballerina/annotation.bal [hunk 2]
```diff
@@ -26,6 +26,16 @@ public type ServiceDescriptorData record {|
 # Service descriptor annotation.
 public annotation ServiceDescriptorData ServiceDescriptor on service;
 
+# Service descriptor data generated at the compile time.
+#
+# + value - Descriptor, which should be set at the compile time
+public type DescriptorData record {|
+    string value = "";
```
Comment (daneshk) [inline ballerina/annotation.bal:33]: shall we make this required? remove default value.

File: ballerina-tests/Ballerina.toml (hunk 1 above)
Comment (dilanSachi) [reply ballerina-tests/Ballerina.toml]: This was a mistake. Fixed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/892/commits/4c9e1de2aabe728a957ef6ef65e5e81aeaeada3a

File: ballerina/annotation.bal (hunk 2 above)
Comment (dilanSachi) [reply ballerina/annotation.bal]: Changed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/892/commits/129750849d669403c1254b21b94cfe738b96872b

**Comment by daneshk** [comment]
@dilanSachi shall we keep a few test cases with the older format. to make sure we are not breaking the existing code.
Others LGTM

**Review by daneshk — APPROVED** [review]

**Comment by dilanSachi** [comment]
> @dilanSachi shall we keep a few test cases with the older format. to make sure we are not breaking the existing code. Others LGTM

Almost all of the runtime tests are in the previous format(centralized descriptor and descriptor map). Will update

**Comment by dilanSachi** [comment]
@daneshk Found 2 issues while updating the runtime tests. Fixed them in the same PR.
https://github.com/ballerina-platform/module-ballerina-grpc/pull/892/commits/3fa9b0c6581a5cab1657613a4e64d71c420c78be
https://github.com/ballerina-platform/module-ballerina-grpc/pull/892/commits/f567107e05672ace93ab4dfe80215032708d6dec

**Pull request merged**